interview-questions.md

```diff
@@ -9,47 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
 Your answer:
+JSX is a hybrid of Javascript and HTML. The difference between them is JSX can use Javascript
+syntax and HTML syntax.
 
 Researched answer:
+JSX is a hybrid between JS and HTML. JSX has HTML-like syntax and 
+was designed with React in mind to make React components more effeciently.
+
+https://stackoverflow.com/questions/33042923/what-is-the-difference-between-javascript-and-jsx#:~:text=JS%20is%20standard%20javascript%2C%20JSX,there's%20not%20much%20else%20there.
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
 Your answer:
+Yarn is a package manager. The JSON file is modified and yarn.lock file is modified.
 
 Researched answer:
+YARN, which stands for Yet Another Resource Negotiator, is a package manager created by Facebook. It has more capabilities, faster,
+safer, and more reliable than than NPM (Node Package Manager). When YARN is ran, it creates a yarn.lock file that lists versions of 
+dependencies that are needed as well as a package.json file that stores the Jest dependency version.
+
+
+https://classic.yarnpkg.com/lang/en/docs/getting-started/#:~:text=Yarn%20is%20a%20package%20manager,t%20ever%20have%20to%20worry.
+https://classic.yarnpkg.com/lang/en/docs/cli/run/#:~:text=yarn%20run%20%5Bscript%5D%20%5B%5D&text=Running%20this%20command%20will%20execute,them%20after%20the%20script%20name.
 
 3. What is a React component?
 
 Your answer:
+A react component is a Javascript file that contains a JSX function. That JSX function can
+be used in App.js using self closing tags.
 
 Researched answer:
+React components are isolated code that act similar to Javascript functions and return HTML.
+React components can be used by placing them in the return of App.js using self closing tags.
+
+https://www.w3schools.com/react/react_components.asp#:~:text=Components%20are%20independent%20and%20reusable,will%20concentrate%20on%20Function%20components.
 
 4. What is the difference between state values and props in React?
 
 Your answer:
+A prop in React is an object that stores a attributes of a tag. State values manage data and state of
+a component.
 
 Researched answer:
+A prop is an object that stores the value of attributes of a tag. State values manange data and the 
+state of the component. When a components state changes the component re-renders.
```

Comment:
I really like that you mentioned that a change in state will cause the component to re-render.